nexus/types/Cargo.toml

```diff
@@ -44,6 +44,7 @@ uuid.workspace = true
 api_identity.workspace = true
 dns-service-client.workspace = true
 gateway-client.workspace = true
+internal-dns.workspace = true
```

Comment:
Is it easy to check if this affects the output of `cargo xtask ls-apis`?  I'm a little worried this is going to make everything that uses `nexus-types` appear to be a client of the DNS servers.  I _think_ that'll be avoided by one of the existing exceptions.

Reply:
`cargo xtask ls-apis apis` showed no difference before and after.

If we're worried about this dependency, it's probably best to move the types out into a separate `internal-dns-types` crate.

Reply:
Ah I see what's going on -- this is because internal-dns (both the types and the resolver) depends on `dns-service-client` for the types. I guess the way to address that is to define the types in another crate, then use `replace`.

Reply:
Good.  I think that as I hoped, this is being covered by this rule:
https://github.com/oxidecomputer/omicron/blob/1a6343e9be47f6c8d2fa4ec4d0d953d0e795deb3/dev-tools/ls-apis/api-manifest.toml#L390-L400

As long as you see no change in the output, there's no (new) problem here.